### ISSUE TITLE
Comparison and benchmarks

### ISSUE THREAD

**Issue by kunaltyagi** [gitsbi/unlib#7]
C++14 compatibility is awesome!!

Are there any table of comparison (feature wise) wrt Boost.Units and nholthaus::units?

Compilation time benchmarks would also be lovely (but not priority).

**Comment by gitsbi** [comment]
No, there currently aren't any such comparisons, and I lack the time and, frankly, motivation, necessary to delve into those other libraries deep enough to make a comparison chart that's not inherently biased toward my library.

There are, however, reasons why I did not use any of those other unit libraries, but decided to write my own.

- [Mateusz Pusz' unit library](https://github.com/mpusz/units) doesn't provide the C++14 compatibility needed for my work. 
- [Boost.Units](https://www.boost.org/doc/libs/1_75_0/doc/html/boost_units.html) allows you to create your own unit system, whereas this library only enable you to use one system: SI. This of course greatly reduces the flexibility of unlib compared to Boost.Units, but makes it easier to use, and faster to compile.
- [Nic Holthaus' unit library](https://github.com/nholthaus/units) has a rather unique way to scale units (think _kilo_, _nano_) which makes it impossible to, for example, write function templates that take any quantity, regardless of scale. However, this was a must-have feature for my work. It also doesn't provide any means for equivalent units that must be distinguished, like _active_, _reactive_, and _apparent_ power in EE.

This list, of course, is _very biased_, because it just lists the criteria used to _dismiss_ the unit libraries we looked at in my job. That, however, is all I have at the moment. But if you, _or anyone else, FTM_, actually know any of those other libraries better than I do (which isn't hard, really), I'd be very interested to hear about more differences. If I have enough of those, I might consider myself unbiased enough to include a comparison chart in the library.
